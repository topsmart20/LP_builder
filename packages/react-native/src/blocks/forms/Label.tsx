import React from 'react';
import { BuilderBlock } from '../../decorators/builder-block.decorator';
import HTML from 'react-native-render-html';

export interface LabelProps {
  attributes?: any;
  text?: string;
  for?: string;
}

@BuilderBlock({
  name: 'Form:Label',
  image:
    'https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2F9322342f04b545fb9a8091cd801dfb5b',
  inputs: [
    {
      name: 'text',
      type: 'text',
      defaultValue: 'Label',
    },
    {
      name: 'for',
      type: 'text',
      helperText: 'The name of the input this label is for',
      advanced: true,
    },
  ],
  noWrap: true,
  // TODO: take inner html or blocsk
  // TODO: optional children? maybe as optional form input
  // that only shows if advanced setting is flipped
  // TODO: defaultChildren
  // canHaveChildren: true,
})
export class Label extends React.Component<LabelProps> {
  render() {
    return <HTML html={this.props.text || ' '} />;
  }
}
