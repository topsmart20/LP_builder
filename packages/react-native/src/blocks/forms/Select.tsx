import React from 'react';
import { BuilderBlock } from '../../decorators/builder-block.decorator';
import { Builder } from '@builder.io/sdk';
import { Picker } from 'react-native';

export interface FormSelectProps {
  options?: { name?: string; value: string }[];
  attributes?: any;
  name?: string;
  value?: string;
  defaultValue?: string;
}

@BuilderBlock({
  name: 'Form:Select',
  image:
    'https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2F83acca093fb24aaf94dee136e9a4b045',
  defaultStyles: {
    alignSelf: 'flex-start',
  },
  inputs: [
    {
      name: 'options',
      type: 'list',
      required: true,
      subFields: [
        {
          name: 'value',
          type: 'text',
          required: true,
        },
        {
          name: 'name',
          type: 'text',
        },
      ],
      defaultValue: [
        {
          value: 'option 1',
        },
        {
          value: 'option 2',
        },
      ],
    },
    {
      name: 'defaultValue',
      type: 'string',
    },
    {
      name: 'value',
      type: 'string',
      advanced: true,
    },
    {
      name: 'name',
      type: 'string',
      // advanced: true,
    },
    {
      name: 'required',
      type: 'boolean',
      defaultValue: false,
    },
  ],
  noWrap: true,
})
export class FormSelect extends React.Component<FormSelectProps> {
  render() {
    const { options } = this.props;
    return (
      <Picker
        value={this.props.value}
        key={Builder.isEditing && this.props.defaultValue ? this.props.defaultValue : 'default-key'}
        defaultValue={this.props.defaultValue}
        name={this.props.name}
        {...this.props.attributes}
      >
        {options &&
          options.map(option => (
            <Picker.Item value={option.value} label={option.name || option.value}></Picker.Item>
          ))}
      </Picker>
    );
  }
}
