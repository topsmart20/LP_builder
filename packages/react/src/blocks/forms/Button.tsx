import React from 'react'
import { BuilderBlock } from '../../decorators/builder-block.decorator'

export interface ButtonProps {
  attributes?: any
  text?: string
}

// TODO: spec all of these as generic builder elements
// and codegen to Vue, React, Angular.........
@BuilderBlock({
  name: 'Form:SubmitButton',
  image:
    'https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2Fdf2820ffed1f4349a94c40b3221f5b98',
  defaultStyles: {
    appearance: 'none',
    paddingTop: '15px',
    paddingBottom: '15px',
    paddingLeft: '25px',
    paddingRight: '25px',
    backgroundColor: '#3898EC',
    color: 'white',
    borderRadius: '4px'
  },
  inputs: [
    {
      name: 'text',
      type: 'text',
      defaultValue: 'Click me'
    }
  ],
  noWrap: true
  // TODO: optional children? maybe as optional form input
  // that only shows if advanced setting is flipped
  // TODO: defaultChildren
  // canHaveChildren: true,
})
export class FormSubmitButton extends React.Component<ButtonProps> {
  render() {
    return (
      <button type="submit" {...this.props.attributes}>
        {this.props.text}
      </button>
    )
  }
}
