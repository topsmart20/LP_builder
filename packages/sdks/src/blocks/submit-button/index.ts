import component from './submit-button-internal.lite';
import { componentInfo } from './component-info';
import { registerComponent } from '../../functions/register-component';
registerComponent(component, componentInfo);
