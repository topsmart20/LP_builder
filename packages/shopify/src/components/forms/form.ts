import { component } from '../../constants/components';
import { blockToLiquid } from '../../functions/block-to-liquid';

export const FormInput = component({
  name: 'Form:Form',
  noWrap: true,
  component: (block, renderOptions, attributes) => {
    const { options } = block.component!;

    return `
      <form
        validate="${options.validate || ''}"
        action="${options.action || ''}"
        method="${options.method || ''}"
        name="${options.name}"
        ${attributes || ''}>
        ${(block.children &&
          block.children.map(child => blockToLiquid(child, renderOptions)).join('\n')) ||
          ''}
      </form>
    `;
  },
});
