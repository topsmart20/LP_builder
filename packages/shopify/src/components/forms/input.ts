import { component } from '../../constants/components';

export const FormInput = component({
  name: 'Form:Input',
  noWrap: true,
  component: (block, _, attributes) => {
    const { options } = block.component!;

    return `
      <input
        placeholder="${options.placeholder || ''}"
        name="${options.name || ''}"
        type="${options.type || ''}"
        value="${options.value || options.defaultValue || ''}"
        ${attributes || ''} />
    `;
  },
});
