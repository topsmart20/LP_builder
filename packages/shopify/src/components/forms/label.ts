import { component } from '../../constants/components';
import { blockToLiquid } from '../../functions/block-to-liquid';

export const Label = component({
  name: 'Form:Label',
  noWrap: true,
  component: (block, renderOptions, attributes) => {
    const { options } = block.component!;

    return `
      <label
        for="${options.for || ''}"
        ${attributes || ''}>
        ${options.text ? `<span class="builder-label-text">${options.text}</span>` : ''}
        ${(block.children &&
          block.children.map(child => blockToLiquid(child, renderOptions)).join('\n')) ||
          ''}
      </label>
    `;
  },
});
