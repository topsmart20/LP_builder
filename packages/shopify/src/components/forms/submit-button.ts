import { component } from '../../constants/components';

export const SubmitButton = component({
  name: 'Form:SubmitButton',
  noWrap: true,
  component: (block, _, attributes) => {
    const { options } = block.component!;

    return `
      <button type="submit" ${attributes || ''}>${options.text || ''}</button>
    `;
  },
});
