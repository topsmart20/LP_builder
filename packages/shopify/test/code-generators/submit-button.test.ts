import { SubmitButton } from '../../src/components/forms/submit-button';
import { el } from '../modules/helpers';

const getSubmitButtonCode = (options: any): string =>
  SubmitButton(
    el({
      component: {
        options,
        name: 'Form:SubmitButton',
      },
    }),
    {}
  );

describe('Submit button code generator', () => {
  test('values for all attributes', () => {
    const code = getSubmitButtonCode({
      text: 'Submit',
    });

    expect(code).toMatchSnapshot();
  });

  test('no values for attributes', () => {
    const code = getSubmitButtonCode({});
    expect(code).toMatchSnapshot();
  });
});
