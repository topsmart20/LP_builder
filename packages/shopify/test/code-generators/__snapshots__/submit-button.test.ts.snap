// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit button code generator no values for attributes 1`] = `
"
      <button type=\\"submit\\" ></button>
    "
`;

exports[`Submit button code generator values for all attributes 1`] = `
"
      <button type=\\"submit\\" >Submit</button>
    "
`;
